plan/plan_to_pb.go

```diff
@@ -89,14 +89,18 @@ func (p *PhysicalTableScan) ToPB(ctx context.Context) (*tipb.Executor, error) {
 // ToPB implements PhysicalPlan ToPB interface.
 func (p *PhysicalIndexScan) ToPB(ctx context.Context) (*tipb.Executor, error) {
 	columns := make([]*model.ColumnInfo, 0, p.schema.Len())
+	tableColumns := p.Table.Cols()
+	if ctx.GetSessionVars().StmtCtx.InUpdateStmt {
```

Comment:
We don't need to check it and get writable columns because all the columns in an index must be public.

Reply:
Yes, now the columns in an index must be public.